code/game/objects/items/tools/weldingtool.dm

```diff
@@ -102,6 +102,15 @@
 	qdel(src)
 
 /obj/item/weldingtool/attack(mob/living/carbon/human/H, mob/user)
+	if(isOn())
```

Comment:
Will fix tomorrow

Reply:
Can you clarify what you mean? I don't understand how this would tie into attackby. attack() already only gets called if attackby returned false. In addition this code block is first because the last part is to do with healing robot body parts, which depends on the target being human, which would break stuff if this code was behind the human check

Reply:
Any day now

Reply:
If the attack would be stopped by a signal or anything else in parent call this shouldn't happen. So you move this to the end of the function and only make this happen if `..()` value is correct.

Reply:
After quick glance at the code this would be pretty awful to do since right now attack doesn't support return values. Adding afterattack_failure() function to call in case the target gets deleted seems more sane.

Reply:
Bruh I just wanted to fix bug, not rework attack code, but I'll see what I can do

Reply:
Okay I changed attack procs but only to the extent of making welders work. I might make the rest of the changes in another PR to avoid getting off track. That satisfactory?

Reply:
This is pretty bad, leaves overridden functions invalid. Honestly i don't quite understand why you insist on adding it in attack, it just makes the whole thing more complicated.

Reply:
Nemvar was the one who told me to move it to attack() in the first place, so this is kind of difficult due to conflicting opinions

Reply:
I am so confused right now, I have two people, one saying "Do X" and the other saying "Do Y"

Reply:
Nemvar is not a maintainer 😏

Reply:
Good point, I'll get this sorted out eventually